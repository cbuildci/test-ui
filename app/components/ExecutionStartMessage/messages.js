/*
 * ExecutionStartMessage Messages
 */

import { defineMessages } from 'react-intl';

const baseStarted = 'Started {relativeCreateTimeHtml}';

export default defineMessages({
    startedWithoutEvent: {
        id: 'app.components.ExecutionStartMessage.startedWithoutEvent',
        defaultMessage: baseStarted,
    },
    startedWithUnknownEvent: {
        id: 'app.components.ExecutionStartMessage.startedWithUnknownEvent',
        defaultMessage: `${baseStarted} after event "{eventType}"`,
    },
    startedForPullRequest: {
        id: 'app.components.ExecutionStartMessage.startedForPullRequest',
        defaultMessage: `${baseStarted} after {action, select,
            opened {{actionHtml}}
            synchronize {{actionHtml} to}
            other {{actionHtml}}
        }
        {pullRequestLinkHtml} by {usernameHtml}`,
    },
    startedForPullRequest_action: {
        id: 'app.components.ExecutionStartMessage.startedForPullRequest_action',
        defaultMessage: `{action, select,
            opened {opening}
            synchronize {push}
            other {action}
        }`,
    },
    startedForPullRequest_link: {
        id: 'app.components.ExecutionStartMessage.startedForPullRequest_link',
        defaultMessage: 'pull request #{pullRequestNumber}',
    },
    startedByUserAction: {
        id: 'app.components.ExecutionStartMessage.startedByUserAction',
        defaultMessage: `${baseStarted} {identifier, select,
            rerun {as {identifierHtml}}
            other {for action {identifierHtml}}
        } initiated by {usernameHtml}`,
    },
    startedByUserAction_identifier: {
        id: 'app.components.ExecutionStartMessage.startedByUserAction_identifier',
        defaultMessage: `{identifier, select,
            rerun {re-run}
            other {{identifier}}
        }`,
    },
});
