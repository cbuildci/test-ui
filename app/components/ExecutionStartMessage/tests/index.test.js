import React from 'react';
import { snapshots } from '../../../../internals/testing/snapshot-util';
import ExecutionStartMessage from '../index';

describe('<ExecutionStartMessage/>', () => {
    it('should render expected JSX with just the create time', () => {
        snapshots(
            <ExecutionStartMessage
                githubHost="git.com"
                owner="foo"
                repo="bar"
                createTime={1500000000000}
            />,
        );
    });

    it('should render expected JSX for unknown event', () => {
        snapshots(
            <ExecutionStartMessage
                githubHost="git.com"
                owner="foo"
                repo="bar"
                createTime={1500000000000}
                event={{
                    event: 'foobar',
                }}
            />,
        );
    });

    it('should render expected JSX for "opened" pull request event', () => {
        snapshots(
            <ExecutionStartMessage
                githubHost="git.com"
                owner="foo"
                repo="bar"
                createTime={1500000000000}
                event={{
                    event: 'pull_request',
                    action: 'opened',
                    pull_request: {
                        number: 5,
                    },
                }}
            />,
        );
    });

    it('should render expected JSX for "synchronize" pull request event', () => {
        snapshots(
            <ExecutionStartMessage
                githubHost="git.com"
                owner="foo"
                repo="bar"
                createTime={1500000000000}
                event={{
                    event: 'pull_request',
                    action: 'synchronize',
                    pull_request: {
                        number: 5,
                    },
                }}
            />,
        );
    });

    it('should render expected JSX for "rerun" user event', () => {
        snapshots(
            <ExecutionStartMessage
                githubHost="git.com"
                owner="foo"
                repo="bar"
                createTime={1500000000000}
                event={{
                    event: 'check_run',
                    action: 'requested_action',
                    requested_action: {
                        identifier: 'rerun',
                    },
                    sender: {
                        login: 'amekkawi-office',
                        type: 'User',
                        id: 13106037,
                    },
                }}
            />,
        );
    });

    it('should render expected JSX for unknown user event', () => {
        snapshots(
            <ExecutionStartMessage
                githubHost="git.com"
                owner="foo"
                repo="bar"
                createTime={1500000000000}
                event={{
                    event: 'check_run',
                    action: 'requested_action',
                    requested_action: {
                        identifier: 'foobar',
                    },
                    sender: {
                        login: 'amekkawi-office',
                        type: 'User',
                        id: 13106037,
                    },
                }}
            />,
        );
    });
});
