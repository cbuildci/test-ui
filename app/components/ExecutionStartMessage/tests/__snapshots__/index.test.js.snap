// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExecutionStartMessage/> should render expected JSX for "opened" pull request event 1`] = `
<FormattedMessage
  defaultMessage="Started {relativeCreateTimeHtml} after {action, select,
            opened {{actionHtml}}
            synchronize {{actionHtml} to}
            other {{actionHtml}}
        }
        {pullRequestLinkHtml}"
  id="app.components.ExecutionStartMessage.startedForPullRequest"
  values={
    Object {
      "action": "opened",
      "actionHtml": <strong>
        <FormattedMessage
          defaultMessage="{action, select,
            opened {opening}
            synchronize {push}
            other {action}
        }"
          id="app.components.ExecutionStartMessage.startedForPullRequest_action"
          values={
            Object {
              "action": "opened",
            }
          }
        />
      </strong>,
      "event": "pull_request",
      "pullRequestLinkHtml": <a
        href="https://git.com/foo/bar/pull/5"
      >
        <FormattedMessage
          defaultMessage="pull request #{pullRequestNumber}"
          id="app.components.ExecutionStartMessage.startedForPullRequest_link"
          values={
            Object {
              "pullRequestNumber": 5,
            }
          }
        />
      </a>,
      "relativeCreateTimeHtml": <strong>
        <InjectIntl(Time)
          date={1500000000000}
        >
          <FormattedRelative
            updateInterval={5000}
            value={1500000000000}
          />
        </InjectIntl(Time)>
      </strong>,
    }
  }
/>
`;

exports[`<ExecutionStartMessage/> should render expected JSX for "opened" pull request event 2`] = `
<span>
  Started 
  <strong>
    <time
      class="position-relative"
      datetime="2017-07-14T02:40:00.000Z"
      id="foobar"
      tabindex="0"
    >
      <span>
        now
      </span>
    </time>
  </strong>
   after 
  <strong>
    <span>
      opening
    </span>
  </strong>
  
        
  <a
    href="https://git.com/foo/bar/pull/5"
  >
    <span>
      pull request #5
    </span>
  </a>
</span>
`;

exports[`<ExecutionStartMessage/> should render expected JSX for "rerun" user event 1`] = `
<FormattedMessage
  defaultMessage="Started {relativeCreateTimeHtml} {identifier, select,
            rerun {as {identifierHtml}}
            other {for action {identifierHtml}}
        } initiated by {usernameHtml}"
  id="app.components.ExecutionStartMessage.startedByUserAction"
  values={
    Object {
      "event": "check_run",
      "identifier": "rerun",
      "identifierHtml": <strong>
        <FormattedMessage
          defaultMessage="{identifier, select,
            rerun {re-run}
            other {{identifier}}
        }"
          id="app.components.ExecutionStartMessage.startedByUserAction_identifier"
          values={
            Object {
              "identifier": "rerun",
            }
          }
        />
      </strong>,
      "relativeCreateTimeHtml": <strong>
        <InjectIntl(Time)
          date={1500000000000}
        >
          <FormattedRelative
            updateInterval={5000}
            value={1500000000000}
          />
        </InjectIntl(Time)>
      </strong>,
      "username": "amekkawi-office",
      "usernameHtml": <a
        href="https://git.com/amekkawi-office"
      >
        @
        amekkawi-office
      </a>,
    }
  }
/>
`;

exports[`<ExecutionStartMessage/> should render expected JSX for "rerun" user event 2`] = `
<span>
  Started 
  <strong>
    <time
      class="position-relative"
      datetime="2017-07-14T02:40:00.000Z"
      id="foobar"
      tabindex="0"
    >
      <span>
        now
      </span>
    </time>
  </strong>
   as 
  <strong>
    <span>
      re-run
    </span>
  </strong>
   initiated by 
  <a
    href="https://git.com/amekkawi-office"
  >
    @amekkawi-office
  </a>
</span>
`;

exports[`<ExecutionStartMessage/> should render expected JSX for "synchronize" pull request event 1`] = `
<FormattedMessage
  defaultMessage="Started {relativeCreateTimeHtml} after {action, select,
            opened {{actionHtml}}
            synchronize {{actionHtml} to}
            other {{actionHtml}}
        }
        {pullRequestLinkHtml}"
  id="app.components.ExecutionStartMessage.startedForPullRequest"
  values={
    Object {
      "action": "synchronize",
      "actionHtml": <strong>
        <FormattedMessage
          defaultMessage="{action, select,
            opened {opening}
            synchronize {push}
            other {action}
        }"
          id="app.components.ExecutionStartMessage.startedForPullRequest_action"
          values={
            Object {
              "action": "synchronize",
            }
          }
        />
      </strong>,
      "event": "pull_request",
      "pullRequestLinkHtml": <a
        href="https://git.com/foo/bar/pull/5"
      >
        <FormattedMessage
          defaultMessage="pull request #{pullRequestNumber}"
          id="app.components.ExecutionStartMessage.startedForPullRequest_link"
          values={
            Object {
              "pullRequestNumber": 5,
            }
          }
        />
      </a>,
      "relativeCreateTimeHtml": <strong>
        <InjectIntl(Time)
          date={1500000000000}
        >
          <FormattedRelative
            updateInterval={5000}
            value={1500000000000}
          />
        </InjectIntl(Time)>
      </strong>,
    }
  }
/>
`;

exports[`<ExecutionStartMessage/> should render expected JSX for "synchronize" pull request event 2`] = `
<span>
  Started 
  <strong>
    <time
      class="position-relative"
      datetime="2017-07-14T02:40:00.000Z"
      id="foobar"
      tabindex="0"
    >
      <span>
        now
      </span>
    </time>
  </strong>
   after 
  <strong>
    <span>
      push
    </span>
  </strong>
   to
        
  <a
    href="https://git.com/foo/bar/pull/5"
  >
    <span>
      pull request #5
    </span>
  </a>
</span>
`;

exports[`<ExecutionStartMessage/> should render expected JSX for unknown event 1`] = `
<FormattedMessage
  defaultMessage="Started {relativeCreateTimeHtml} after event \\"{event}\\""
  id="app.components.ExecutionStartMessage.startedWithUnknownEvent"
  values={
    Object {
      "event": "foobar",
      "relativeCreateTimeHtml": <strong>
        <InjectIntl(Time)
          date={1500000000000}
        >
          <FormattedRelative
            updateInterval={5000}
            value={1500000000000}
          />
        </InjectIntl(Time)>
      </strong>,
    }
  }
/>
`;

exports[`<ExecutionStartMessage/> should render expected JSX for unknown event 2`] = `
<span>
  Started 
  <strong>
    <time
      class="position-relative"
      datetime="2017-07-14T02:40:00.000Z"
      id="foobar"
      tabindex="0"
    >
      <span>
        now
      </span>
    </time>
  </strong>
   after event "foobar"
</span>
`;

exports[`<ExecutionStartMessage/> should render expected JSX for unknown user event 1`] = `
<FormattedMessage
  defaultMessage="Started {relativeCreateTimeHtml} {identifier, select,
            rerun {as {identifierHtml}}
            other {for action {identifierHtml}}
        } initiated by {usernameHtml}"
  id="app.components.ExecutionStartMessage.startedByUserAction"
  values={
    Object {
      "event": "check_run",
      "identifier": "foobar",
      "identifierHtml": <strong>
        <FormattedMessage
          defaultMessage="{identifier, select,
            rerun {re-run}
            other {{identifier}}
        }"
          id="app.components.ExecutionStartMessage.startedByUserAction_identifier"
          values={
            Object {
              "identifier": "foobar",
            }
          }
        />
      </strong>,
      "relativeCreateTimeHtml": <strong>
        <InjectIntl(Time)
          date={1500000000000}
        >
          <FormattedRelative
            updateInterval={5000}
            value={1500000000000}
          />
        </InjectIntl(Time)>
      </strong>,
      "username": "amekkawi-office",
      "usernameHtml": <a
        href="https://git.com/amekkawi-office"
      >
        @
        amekkawi-office
      </a>,
    }
  }
/>
`;

exports[`<ExecutionStartMessage/> should render expected JSX for unknown user event 2`] = `
<span>
  Started 
  <strong>
    <time
      class="position-relative"
      datetime="2017-07-14T02:40:00.000Z"
      id="foobar"
      tabindex="0"
    >
      <span>
        now
      </span>
    </time>
  </strong>
   for action 
  <strong>
    <span>
      foobar
    </span>
  </strong>
   initiated by 
  <a
    href="https://git.com/amekkawi-office"
  >
    @amekkawi-office
  </a>
</span>
`;

exports[`<ExecutionStartMessage/> should render expected JSX with just the create time 1`] = `
<FormattedMessage
  defaultMessage="Started {relativeCreateTimeHtml}"
  id="app.components.ExecutionStartMessage.startedWithoutEvent"
  values={
    Object {
      "relativeCreateTimeHtml": <strong>
        <InjectIntl(Time)
          date={1500000000000}
        >
          <FormattedRelative
            updateInterval={5000}
            value={1500000000000}
          />
        </InjectIntl(Time)>
      </strong>,
    }
  }
/>
`;

exports[`<ExecutionStartMessage/> should render expected JSX with just the create time 2`] = `
<span>
  Started 
  <strong>
    <time
      class="position-relative"
      datetime="2017-07-14T02:40:00.000Z"
      id="foobar"
      tabindex="0"
    >
      <span>
        now
      </span>
    </time>
  </strong>
</span>
`;
